### DISCUSSION TITLE
Delete the `bash.exe` alias in System32 AND WindowsApps folders

### DISCUSSION THREAD

**Discussion by julianxhokaxhiu** [microsoft/WSL#9681] [Q&A]
Hello,

I'm trying to understand how can I delete the `bash.exe` alias after I enabled and installed WSL on Windows 11. This alias [seems currently deprecated](https://learn.microsoft.com/en-us/windows/wsl/troubleshooting?source=recommendations#using-older-versions-of-windows-and-wsl) and I need to delete it because of a current conflict with vcpkg ( see https://github.com/microsoft/vcpkg/issues/29658#issuecomment-1436021393 ).

Simply deleting it although from explorer won't work, throwing an error `0x80070780`. I did try to search online but I wasn't able to find anything related. Can anyone please help me figure out how to delete it **while preserving the WSL installation**?

Thank you in advance for any help given,
Julian

**Comment by Biswa96** [comment]
Instead of deleting system file, you should remove some paths from PATH environment variable. When you want to use bash.exe other than WSL's one, try to prepend the directory path instead of appending it. For example, using command like this `set PATH=C:\path\to\other\dir;%PATH%`

**Reply by julianxhokaxhiu** [reply]
I wish it was that easy. If you look at the issue that's exactly the proposed solution but it won't come anytime soon and I am blocked atm. This is why I'm asking for this other option. Removing the PATHs from my own env is not an option either as I need it for other purposes. Thanks anyway.

**Reply by jonmoore** [reply]
Removing c:\windows\system32 from PATH is hardly practical, while fiddling with PATH to prepend user-specified directories goes hard against the Windows default behavior of placing the system PATH before the user one.

**Reply by felipec** [reply]
Removing that location from the `PATH` is not going to help with functions like `CreateProcess` which gives priority to the 32-bit system directory.

**Reply by ReuelAlbert-Dev** [reply]
the only way is to fix bugs and improve

**Reply by felipec** [reply]
@ReuelAlbert-Dev unfortunately it doesn't seem Microsoft is interested in fixing this bug: #13277.

**Comment by elsaco** [comment]
```
PS C:\Windows\System32> cacls c:\windows\system32\bash.exe
c:\windows\system32\bash.exe NT SERVICE\TrustedInstaller:F
                             BUILTIN\Administrators:R
                             NT AUTHORITY\SYSTEM:R
                             BUILTIN\Users:R
                             APPLICATION PACKAGE AUTHORITY\ALL APPLICATION PACKAGES:R
                             APPLICATION PACKAGE AUTHORITY\ALL RESTRICTED APPLICATION PACKAGES:R
```

Bash launcher is owned by the `TrustedInstaller` and that makes it difficult to delete. It can be nuked! Use `takeown` and `cacls` utils or boot with a live Linux distro, mount the partition and delete `bash.exe`.

 ⚠️ It might nuke your system also ⚠️

**Reply by julianxhokaxhiu** [reply]
Yeah I thought it would have been about permissions...couldn't this be provided as part of the WSL solution? Like I want WSL but I don't want to have `bash.exe` as an alias.

**Reply by jonmoore** [reply]
Seems like the best of a bad situation.
If one doesn't need WSL then this will work (1 on its own is insufficient)
1. Remove WSL from "Add or remove programs"
2. Deselect "Windows Subsystem for Linux" in "Turn Windows features on or off"
3. Restart Windows

It's common to run bash inside standard (non-WSL) Windows and the Windows and WSL devs must know this. I find it really unprofessional of Microsoft to treat a critical item like C:\Windows\System32 so casually.

**Reply by Ashark** [reply]
Thanks for the advice, it worked.
```
icacls "C:\Windows\System32\bash.exe"
takeown /f "C:\Windows\System32\bash.exe" /a
icacls "C:\Windows\System32\bash.exe" /grant Administrators:F
```
But still, it did not allow to delete that:
```
del "C:\Windows\System32\bash.exe"
Remove-Item: Access to the path 'C:\Windows\System32\bash.exe' is denied.
```
So, in the gui in explorer, in the properties for the bash.exe file, I saw that Administrator still does not have _full_ permissions. I checked checkbox to grant that permission. And then I was able to delete the file.
